uaclient/contract.py

```diff
@@ -295,17 +295,19 @@ def request_updated_contract(
                 allow_enable=allow_enable,
             )
         except exceptions.UserFacingError as e:
-            user_errors.append(e)
+            with util.disable_log_to_console():
+                logging.exception(str(e))
+            user_errors.append(status.MESSAGE_ATTACH_FAILURE_DEFAULT_SERVICES)
         except Exception as e:
             with util.disable_log_to_console():
                 logging.exception(str(e))
-            raise exceptions.UserFacingError(
-                "Unexpected error handling Ubuntu Advantage contract changes"
+            user_errors.append(
+                status.MESSAGE_UNEXPECTED_ERROR_DURING_OP_TMPL.format(
+                    operation="attach"
+                )
             )
     if user_errors:
-        raise exceptions.UserFacingError(
-            status.MESSAGE_ATTACH_FAILURE_DEFAULT_SERVICES
-        )
+        raise exceptions.UserFacingError(user_errors[0])
```

Comment:
As written, I think this will only display the "unexpected error" message if it is the first error we encounter. Do you think we should instead always emit that message if _any_ of the errors are unexpected?

Reply:
I've adjusted this logic to insert MESSAGE_UNEXPECTED_ERROR_DURING_OP_TMPL message to the first item in the user_errors list. This way we will raise the unexpected error message to the user even if both types of errors have occurred.

Reply:
Do we need to put the error in `user_errors` at all? It seems to me that we could just have a bool that tracks if we've seen an "unexpected" error; if we have, we emit the appropriate message, if we haven't, we emit `user_errors[0]`.

(And, actually, in that scenario I think all the elements in `user_errors` must be `status.MESSAGE_ATTACH_FAILURE_DEFAULT_SERVICES`, so perhaps we can replace it entirely with two bools?)